### ISSUE TITLE
Add vector-tile `id` to cluster features to work with GL JS v0.46.0 FeatureStates

### ISSUE THREAD

**Issue by ryanbaumann** [mapbox/geojson-vt#106]
# Problem

GL JS v0.46.0 launched the setFeatureState() API, which allows a feature to be updated at run-time using the tile feature index.

# Solution

Add the `cluster_id` as a vector tile feature `id`, allowing clusters to be used with setFeatureState for hover and highlight interactions with cluster layers in GL JS.

cc/ @mourner

**Comment by ryanbaumann** [comment]
Linking to https://github.com/mapbox/mapbox-gl-js/issues/6849 as the GL-JS specific issue.

**Comment by mourner** [comment]
@ryanbaumann did you mean to post clustering-related improvements to the supercluster repo? `geojson-vt` only deals with unclustered sources.

**Comment by ryanbaumann** [comment]
@mourner yes, thanks for the catch!  Supercluster issue -> https://github.com/mapbox/supercluster/issues/88

Please leave this open if we would potentially handle geojson feature `id` creation in `geojson-vt`.  Otherwise, feel free to close.
